### DISCUSSION TITLE
Implement React Query SSR with create react app?

### DISCUSSION THREAD

**Discussion by Uttu316** [TanStack/query#3263] [Q&A]
How  React query supports SSR in create react app not Next js.

Any blog post or tutorial which can help me to add SSR support of react query in create react app.

**Comment by TkDodo** [comment]
crate react app, as far as I know, is for client side rendering only. see: https://reactjs.org/docs/create-a-new-react-app.html#recommended-toolchains

**Reply by Uttu316** [reply]
My project started from create react app and currently it is in between of development and there is no plan to migrate to Nextjs.
I am adding SSR support in create react app.Hence, I need to add SSR support for React query as well.

Is there any way I can implement this?

**Reply by TkDodo** [reply]
how does SSR work with create-react-app _without_ react-query? I've never heard of anyone implementing SSR with CRA 🤷 

Here is a generic guide on how to make SSR work without any framework, so likely you can build on top of that: https://react-query.tanstack.com/guides/ssr#using-other-frameworks-or-custom-ssr-frameworks
